Automattic/jetpack
# Actions: Build master-stable artifact (zip)

#### Changes proposed in this Pull Request:

Part of an overall effort #14589

This changes the behavior of build-master.yml workflow to upload an [artifact](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/persisting-workflow-data-using-artifacts) of the build rather than pushing to another branch. I'm open to discussion whether we will still need a branch for this. If so, we can add it in the same workflow. 

Also with this PR is a new (very simple) script that will purge dev files. My hope is that we can completely replace .svnignore with this file eventually. 

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Updates existing action 

#### Testing instructions:






The way I was testing was by manually changing the target branch at the top of the workflow to point to this one, and then pushing a commit to it. Then you can watch the workflow magic happen in https://github.com/Automattic/jetpack/actions. 

You can see the artifact build in the dropdown. Install that on a site. 

- Check out the build size. I'm not convinced it's actually compressing on zip, which might be fine. 
- You shouldn't see any dev files in there at all. 

#### Proposed changelog entry for your changes:

N/A
